### PR TITLE
major changes

### PR DESCRIPTION
**Pull request by lucasborin**

### CODE REVIEW

File: clean_code_main/clean_code_checks/y_check_comment_usage.clas.abap [hunk 1]
```diff
@@ -41,10 +41,12 @@ CLASS Y_CHECK_COMMENT_USAGE IMPLEMENTATION.
...
   METHOD checkif_error.
-    DATA(check_configuration) = detect_check_configuration( threshold = round( val =  percentage_of_comments
-                                                                               dec = 0
-                                                                               mode = cl_abap_math=>round_down )
-                                                            include = get_include( p_level = statement_for_message-level ) ).
+
+    DATA(check_configuration) = detect_check_configuration( error_count = round( val =  percentage_of_comments
+                                                                                 dec = 0
+                                                                                 mode = cl_abap_math=>round_down )
+                                                            statement = statement_for_message ).
```
Comment (lucasborin) [inline clean_code_main/clean_code_checks/y_check_comment_usage.clas.abap:48]: Can we remove the IF from line 54 as the threshold is now being checked in the detect_check_configuration?

File: clean_code_main/clean_code_checks/y_check_equals_sign_chaining.clas.abap [hunk 2]
```diff
@@ -28,7 +28,7 @@ CLASS Y_CHECK_EQUALS_SIGN_CHAINING IMPLEMENTATION.
     settings-pseudo_comment = '"#EC EQUALS_CHAINING'.
     settings-documentation = |{ c_docs_path-checks }equals-sign-chaining.md|.
     settings-disable_threshold_selection = abap_true.
-    settings-threshold = 1.
+    settings-threshold = 0.
```
Comment (lucasborin) [inline clean_code_main/clean_code_checks/y_check_equals_sign_chaining.clas.abap:31]: Is there any impact on customers who are using this check already?

File: clean_code_main/clean_code_checks/y_check_is_interface_in_class.clas.abap [hunk 3]
```diff
@@ -30,7 +30,7 @@ CLASS Y_CHECK_IS_INTERFACE_IN_CLASS IMPLEMENTATION.
 
     settings-pseudo_comment = '"#EC INTF_IN_CLASS' ##NO_TEXT.
     settings-disable_threshold_selection = abap_true.
-    settings-threshold = 0.
+    settings-threshold = 1.
```
Comment (lucasborin) [inline clean_code_main/clean_code_checks/y_check_is_interface_in_class.clas.abap:33]: Any impact on customers who are using this check already?

File: clean_code_main/clean_code_checks/y_check_method_output_param.clas.abap [hunk 4]
```diff
@@ -127,17 +131,20 @@ CLASS Y_CHECK_METHOD_OUTPUT_PARAM IMPLEMENTATION.
...
+    IF has_error( ) = abap_false.
       RETURN.
     ENDIF.
 
-    IF calculate_param_combination( ) > check_configuration-threshold.
-      raise_error( statement_level     = statement-level
-                   statement_index     = index
-                   statement_from      = statement-from
-                   error_priority      = check_configuration-prio ).
+    DATA(check_configuration) = detect_check_configuration( statement ).
```
Comment (lucasborin) [inline clean_code_main/clean_code_checks/y_check_method_output_param.clas.abap:138]: Shouldn't we set the default threshold to 0 as it is being done in the check_*_usage checks?

File: clean_code_main/clean_code_checks/y_check_equals_sign_chaining.clas.abap (hunk 2 above)
Comment (eugen-guenther-sap) [reply clean_code_main/clean_code_checks/y_check_equals_sign_chaining.clas.abap:31]: looks ok

File: clean_code_main/clean_code_checks/y_check_is_interface_in_class.clas.abap (hunk 3 above)
Comment (eugen-guenther-sap) [reply clean_code_main/clean_code_checks/y_check_is_interface_in_class.clas.abap:33]: no

File: clean_code_main/clean_code_checks/y_check_method_output_param.clas.abap (hunk 4 above)
Comment (eugen-guenther-sap) [reply clean_code_main/clean_code_checks/y_check_method_output_param.clas.abap:138]: the check did not use a threshold at all.

File: clean_code_main/clean_code_checks/y_check_num_output_parameter.clas.testclasses.abap [hunk 5]
```diff
@@ -195,21 +196,24 @@ CLASS local_test_class IMPLEMENTATION.
...
+    assert_warnings( 2 ).
     assert_pseudo_comments( 0 ).
   ENDMETHOD.
 
   METHOD pseudo_comment_ok.
     ref_scan_manager_double->set_pseudo_comment_ok( ).
     cut->run( ).
     assert_errors( 0 ).
-    assert_pseudo_comments( 4 ).
+    assert_warnings( 1 ).
```
Comment (eugen-guenther-sap) [inline clean_code_main/clean_code_checks/y_check_num_output_parameter.clas.testclasses.abap:215]: why is a warning in the pseudo comment section?

**Pull request merged**